File: 0160 - Intersection of Two Linked Lists/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, A: ListNode, B: ListNode) -> Optional[ListNode]:
        nowA, nowB = A, B
        while nowA != nowB:
            nowA = nowA.next if nowA else B
            nowB = nowB.next if nowB else A
        return nowA

File: 0160 - Intersection of Two Linked Lists/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *A, ListNode *B) {
        ListNode *nowA = A, *nowB = B;
        while(nowA != nowB){
            nowA = nowA?nowA->next:B;
            nowB = nowB?nowB->next:A;
        }
        return nowA;
    }
};